### ISSUE TITLE
modern light theme: secondary views with wrong contrast

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/eclipse.platform.ui#2200]
As far as i understand every View but the Editor is supposed to have low contrast but currently it is  a wild mixture of shades

Examples:

JUnit Stack trace part:
![image](https://github.com/user-attachments/assets/95fc91cd-ae36-426e-8ec8-5c8b8355d9ae)
History VIew:
![image](https://github.com/user-attachments/assets/45d13115-2e1a-4d03-88ac-d371e218c246)
Problem Views, Error Log  Title:
![image](https://github.com/user-attachments/assets/7d2b5898-fcba-46ab-b1e1-94ee6ceff600)
![image](https://github.com/user-attachments/assets/9206345c-c65f-46e7-819d-34e8b8f28574)
Console:
![image](https://github.com/user-attachments/assets/59175840-0b9b-42f2-8d8b-4d3311fb9ec3)

Also the items in the Problems View have even are darker background then the other views
![image](https://github.com/user-attachments/assets/2bdaa2a3-c090-4fc2-8db6-d98206aecdf9)

**Comment by merks** [comment]
For EMF-based editors, the properties view is a principle focal point when editing a model.  Here is how things used to look:

![image](https://github.com/user-attachments/assets/f6b7d26f-d68a-4b79-80cd-0c70da6bbf81)

This is how it looks now:

![image](https://github.com/user-attachments/assets/6a85cc22-45b2-4177-8689-6df77c59705d)

It's hard to tell what really has focus because there's always a blue under/over-line.  The underlining of the tabbed page in the eidtor is different from the rest of the underlining, and changes color depending on focus.  The breadcrumbs have some white stuff in them that I never seen before.  Where did that come from?

Ranting constructively, this is one of the things that drive me to despair with theming.  I might set a color, but something else will change that color and in the end I have no clue of how I can set colors such that all these different themes won't produce something completely unreadable in some cases.  Basically one has to test every editor, view, and dialog on every theme, and even then I'm often left wondering, how do I fix this?

**Comment by BeckerWdf** [comment]
> The breadcrumbs have some white stuff in them that I never seen before. Where did that come from?

@mvm-sap: Can you answer this question?

**Comment by BeckerWdf** [comment]
> Ranting constructively

I like that term ;-)

> this is one of the things that drive me to despair with theming. I might set a color, but something else will change that color and in the end I have no clue of how I can set colors such that all these different themes won't produce something completely unreadable in some cases. Basically one has to test every editor, view, and dialog on every theme, and even then I'm often left wondering, how do I fix this?

I really agree.

**Comment by akurtakov** [comment]
That's the biggest issue with theming - one either has to set *EVERY* color everywhere or such discrepancies will happen. It's one of the reasons why SWT by default tries to not hardcode colors but rather use the respective OS defined colors as we would most likely never have the manpower and designing capabilities of an OS developing company.

**Comment by mvm-sap** [comment]
> The breadcrumbs have some white stuff in them that I never seen before. Where did that come from?

Yes, we wanted the background of breadcrumbs to have the same color as the text editor, hence set the background of composites in the compatibility editor to white. But, each item in the breadcrumb is not picking that as jdt has set to grey in their theming extension. I can revert those changes, so that it is completely grey and each plugin has to customize it with white background for breadcrumbs in their respective CSS extension

**Comment by BeckerWdf** [comment]
> For EMF-based editors, the properties view is a principle focal point when editing a model. Here is how things used to look:
> 
> ![image](https://private-user-images.githubusercontent.com/208716/358494424-f6b7d26f-d68a-4b79-80cd-0c70da6bbf81.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwNzYwMzYsIm5iZiI6MTcyNzA3NTczNiwicGF0aCI6Ii8yMDg3MTYvMzU4NDk0NDI0LWY2YjdkMjZmLWQ2OGEtNGI3OS04MGNkLTBjNzBkYTZiYmY4MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyM1QwNzE1MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OGNkMDM5NTZhMzYwMWY2N2Q0ZWY4MWFjNDRhOTQzNGIyOTU5NTA4NzNlZDA5YTExNDJmNzI4YThiZjkzMzM0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xeGO_UtXBcEFxCr4cjdtC635EDeMqYRpEahkne37JIE)
> 
> This is how it looks now:
> 
> ![image](https://private-user-images.githubusercontent.com/208716/358494832-6a85cc22-45b2-4177-8689-6df77c59705d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwNzYwMzYsIm5iZiI6MTcyNzA3NTczNiwicGF0aCI6Ii8yMDg3MTYvMzU4NDk0ODMyLTZhODVjYzIyLTQ1YjItNDE3Ny04Njg5LTZkZjc3YzU5NzA1ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyM1QwNzE1MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMDA3NWUzZWQ1NDg5OTRlZjY0ODRjYWVjNDE1Y2I3YzdiY2NhZmI0NzRiZjBhMDMyNjlhY2VkMzYyYWYzMzFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.UmlWt5HUqaRhLPbGBbsz6zdLT-DcPF7Ya35kBfnUX5g)
> 
> It's hard to tell what really has focus because there's always a blue under/over-line. The underlining of the tabbed page in the eidtor is different from the rest of the underlining, and changes color depending on focus. The breadcrumbs have some white stuff in them that I never seen before. Where did that come from?
> 
> Ranting constructively, this is one of the things that drive me to despair with theming. I might set a color, but something else will change that color and in the end I have no clue of how I can set colors such that all these different themes won't produce something completely unreadable in some cases. Basically one has to test every editor, view, and dialog on every theme, and even then I'm often left wondering, how do I fix this?

@merks: Can you pls. verify again with today's version of the "Light (Preview)" theme? Are the issues still there?

**Comment by jukzi** [comment]
Type hierarchy: same error:
![image](https://github.com/user-attachments/assets/19c65d67-5d70-4340-86ba-d304f019b9d4)
